pkg/scanners/template/scanner_template.go

```diff
@@ -0,0 +1,145 @@
+package template
+
+import (
+	"context"
+	"io"
+	"os"
+	"os/signal"
+	"syscall"
+
+	eraserv1alpha1 "github.com/Azure/eraser/api/v1alpha1"
+	"github.com/go-logr/logr"
+	"golang.org/x/sys/unix"
+
+	"github.com/Azure/eraser/pkg/metrics"
+	util "github.com/Azure/eraser/pkg/utils"
+	"go.opentelemetry.io/otel/metric/global"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+type ImageProvider interface {
+	Initialize() []eraserv1alpha1.Image
+	SendToEraser(vulnerableImages, failedImages []eraserv1alpha1.Image)
+	Cleanup()
```

Comment:
```suggestion
	ReceiveImages() []eraserv1alpha1.Image
	SendImages(vulnerableImages, failedImages []eraserv1alpha1.Image)
	Finish()
```
A couple of name suggestions. Since this is a public interface, some code documentation is appropriate here. Documenting the interface as a whole with what it does, then add a doc comment for each function, giving a brief explanation of what it does and how it fits in the context of the pipeline.

Finally, it should be noted that calling `Cleanup()` is *required* after the scanning is complete.